tests/integration_test.go

```diff
@@ -721,9 +721,13 @@ func TestError_ShouldOnlyTakeOfferOnce(t *testing.T) {
 	require.Equal(t, 1, len(providers))
 	require.GreaterOrEqual(t, len(providers[0]), 2)
 
-	errCh := make(chan error)
+	errCh := make(chan error, 2)
+
+	var wg sync.WaitGroup
+	wg.Add(2)
```

Comment:
It was this test that was occasionally causing `TestSuccess_ConcurrentSwaps` after it to fail. The second attempt to take the offer was sometimes still alive after the test completed and recreated the offer ID here:
https://github.com/noot/atomic-swap/blob/master/protocol/xmrmaker/swap_state.go#L178-L189